lang/src/org/partiql/lang/eval/physical/operators/DefaultOperatorCompilers.kt

```diff
@@ -0,0 +1,298 @@
+import org.partiql.lang.eval.relation.RelationScope
+import org.partiql.lang.eval.relation.RelationType
+import org.partiql.lang.eval.relation.relation
+import org.partiql.lang.eval.sourceLocationMeta
+import org.partiql.lang.eval.unnamedValue
+import org.partiql.lang.util.toIntExact
+
+// DL TODO: terminology everywhere.  For instance, I don't know if I like the term operator "compiler" applied here
+
+val DEFAULT_OPERATOR_COMPILERS = OperatorCompilerCatalog(
+    listOf(
+        createPhysicalOperatorCompiler<PartiqlPhysical.Bexpr.Scan>(DEFAULT_IMPL_NAME) { node, compiler, valueFactory ->
+            val exprThunk = compiler.compile(node.expr)
+
+            val asIndexSetter = node.asDecl.toSetVariableFunc()
+            val atIndexSetter = node.atDecl?.toSetVariableFunc()
+            val byIndexSetter = node.byDecl?.toSetVariableFunc()
+
+            relationThunk(node.metas) { state ->
+                val valueToScan = exprThunk.invoke(state)
+
+                // coerces non-collection types to a singleton Sequence<>.
+                val rows: Sequence<ExprValue> = when (valueToScan.type) {
+                    ExprValueType.LIST, ExprValueType.BAG -> valueToScan.asSequence()
+                    else -> sequenceOf(valueToScan)
+                }
+
+                relation(RelationType.BAG) {
+                    val rowsIter: Iterator<ExprValue> = rows.iterator()
+                    while (rowsIter.hasNext()) {
+                        val item = rowsIter.next()
+
+                        // Note that although we could access [EvaluatorState.registers] directly within relationThunk,
+                        // we don't as a means to demonstrate the proper way that custom operator implementations should
+                        // set the values of variables.  See [toSetFunc] and [SetVarFunc] for more info.
+                        asIndexSetter(state, item.unnamedValue()) // Remove any ordinal (output is a bag)
+
+                        if (atIndexSetter != null) {
+                            atIndexSetter(state, item.name ?: valueFactory.missingValue)
+                        }
+
+                        if (byIndexSetter != null) {
+                            byIndexSetter(state, item.address ?: valueFactory.missingValue)
+                        }
+                        yield()
+                    }
+                }
+            }
+        },
+        createPhysicalOperatorCompiler<PartiqlPhysical.Bexpr.Filter>(DEFAULT_IMPL_NAME) { node, compiler, _ ->
+            val predicateThunk = compiler.compile(node.predicate)
+            val sourceThunk = compiler.compile(node.source)
+
+            relationThunk(node.metas) { env ->
+                val sourceToFilter = sourceThunk(env)
+                createFilterRelItr(sourceToFilter, predicateThunk, env)
+            }
+        },
+        createPhysicalOperatorCompiler<PartiqlPhysical.Bexpr.Join>(DEFAULT_IMPL_NAME) { node, compiler, valueFactory ->
```

Comment:
Consider adding a `valueFactory` to the `EvaluatorState` class so it doesn't need to be passed separately here.